src/test/java/com/uber/cadence/worker/StickyWorkerTest.java

```diff
@@ -0,0 +1,325 @@
+    // Arrange
+    String taskListName = "queryStickyTest";
+    TestEnvironmentWrapper wrapper =
+        new TestEnvironmentWrapper(
+            new Worker.FactoryOptions.Builder().setEnableStickyExecution(true).Build());
+    Worker.Factory factory = wrapper.getWorkerFactory();
+    Worker worker = factory.newWorker(taskListName);
+    worker.registerWorkflowImplementationTypes(GreetingWorkflowImpl.class);
+    factory.start();
+
+    WorkflowOptions workflowOptions =
+        new WorkflowOptions.Builder()
+            .setTaskList(taskListName)
+            .setExecutionStartToCloseTimeout(Duration.ofDays(30))
+            .setTaskStartToCloseTimeout(Duration.ofSeconds(30))
+            .build();
+    GreetingWorkflow workflow =
+        wrapper.getWorkflowClient().newWorkflowStub(GreetingWorkflow.class, workflowOptions);
+
+    // Act
+    WorkflowClient.start(workflow::getGreeting);
+
+    Thread.sleep(200); // Wait for workflow to start
+
+    DeciderCache cache = factory.getCache();
+    assertNotNull(cache);
+    assertEquals(1, cache.size());
+
+    // Assert
+    assertEquals(workflow.getProgress(), GreetingWorkflow.Status.WAITING_FOR_NAME);
+
+    workflow.waitForName("World");
+    String greeting = workflow.getGreeting();
+
+    assertEquals("Hello World!", greeting);
+    assertEquals(workflow.getProgress(), GreetingWorkflow.Status.GREETING_GENERATED);
+    wrapper.close();
+  }
+
+  @Test
+  public void workflowsCanBeQueriedAfterEviction() throws Exception {
+    // Arrange
+    String taskListName = "queryEvictionStickyTest";
+    TestEnvironmentWrapper wrapper =
+        new TestEnvironmentWrapper(
+            new Worker.FactoryOptions.Builder().setEnableStickyExecution(true).Build());
+    Worker.Factory factory = wrapper.getWorkerFactory();
+    Worker worker = factory.newWorker(taskListName);
+    worker.registerWorkflowImplementationTypes(GreetingWorkflowImpl.class);
+    factory.start();
+
+    WorkflowOptions workflowOptions =
+        new WorkflowOptions.Builder()
+            .setTaskList(taskListName)
+            .setExecutionStartToCloseTimeout(Duration.ofDays(30))
+            .setTaskStartToCloseTimeout(Duration.ofSeconds(30))
+            .build();
+    GreetingWorkflow workflow =
+        wrapper.getWorkflowClient().newWorkflowStub(GreetingWorkflow.class, workflowOptions);
+
+    // Act
+    WorkflowClient.start(workflow::getGreeting);
+
+    Thread.sleep(200); // Wait for workflow to start
+
+    DeciderCache cache = factory.getCache();
+    assertNotNull(cache);
+    assertEquals(1, cache.size());
+    cache.invalidateAll();
+    assertEquals(0, cache.size());
+
+    // Assert
+    assertEquals(workflow.getProgress(), GreetingWorkflow.Status.WAITING_FOR_NAME);
+
+    workflow.waitForName("World");
+    String greeting = workflow.getGreeting();
+
+    assertEquals("Hello World!", greeting);
+    assertEquals(workflow.getProgress(), GreetingWorkflow.Status.GREETING_GENERATED);
+    wrapper.close();
+  }
+
```

Comment:
Let's add the test for the sticky task schedule to start timeout.

Reply:
There is one already. Take a look at ReplayDeciderTaskHandlerTests.ifStickyExecutionAttributesAreSetThenWorkflowsAreCached()